server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -98,6 +98,7 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion UPGRADE_LUCENE_9_9_1 = def(8_500_008, Version.LUCENE_9_9_1);
     public static final IndexVersion ES_VERSION_8_13 = def(8_500_009, Version.LUCENE_9_9_1);
     public static final IndexVersion NEW_INDEXVERSION_FORMAT = def(8_501_00_0, Version.LUCENE_9_9_1);
+    public static final IndexVersion UPGRADE_LUCENE_9_9_2 = def(8_501_00_1, Version.LUCENE_9_9_2);
```

Comment:
This is very timely - I've got some open PRs to redefine the index version constants, in lieu of a patch version - see https://github.com/elastic/elasticsearch/issues/104620#issuecomment-1910040975.

We're aided here by 8.13 not being released yet, and serverless completely moved off the 8.12 branches. So we can kinda do things independently for main and 8.12, as long as it all lines back up at the end.

Once #104754 goes in, this then gives us the space to add new version constants for use by 8.12. So we can define 8500010 as the index version with the new lucene, and a new constant 8502000 for `main` for the new lucene version. This will then be picked up by serverless.

So we'll then have:

- 8500008 - 8.12.0, lucene 9.9.1
- 8500009 - 8.12.1 (temp), lucene 9.9.1
- 8500010 - 8.12.1 release, lucene 9.9.2
- 8501000 - 8.13.0, lucene 9.9.1
- 8502000 - 8.13.0, lucene 9.9.2

Serverless will now all be on 8501000, so it only go to 8502000. The 8.12/8.13 releases will never see version 8501000, as that is not used in any released version, it'll effectively be hidden. The index version will go 8500008(9.9.1) -> 8500010(9.9.2) -> 8502000(9.9.2)

Reply:
That all sounds great. Thanks for the details analysis and guidance.  So this PR (and the backport of 9.9.2 to 8.12.1) are now dependent on #104754 - which seems fine.  Once #104754 is merged everything will just "fall out in the wash!" - work out seamlessly. :-)

Reply:
To be clear, there's no reason to use a patch version here - this can just be version `8_502_00_0`

Reply:
Thanks, that makes sense. Done